bindings/go/descriptor/runtime/descriptor.go

```diff
@@ -0,0 +1,281 @@
+type ResourceRelation string
+
+const (
+	// LocalRelation defines a internal relation
+	// which describes a internally maintained resource in the origin's context.
+	LocalRelation ResourceRelation = "local"
+	// ExternalRelation defines a external relation
+	// which describes a resource maintained by a third party vendor in the origin's context.
+	ExternalRelation ResourceRelation = "external"
+)
+
+// A Resource is a delivery artifact, intended for deployment into a runtime environment, or describing additional content,
+// relevant for a deployment mechanism.
+// For example, installation procedures or meta-model descriptions controlling orchestration and/or deployment mechanisms.
+// See https://github.com/open-component-model/ocm-spec/blob/main/doc/01-model/02-elements-toplevel.md#resources
+// +k8s:deepcopy-gen=true
+type Resource struct {
+	ElementMeta `json:",inline"`
+	// SourceRefs defines a list of source names.
+	// These entries reference the sources defined in the
+	// component.sources.
+	SourceRefs []SourceRef `json:"-"`
+	// Type describes the type of the object.
+	Type string `json:"-"`
+	// Relation describes the relation of the resource to the component.
+	// Can be a local or external resource.
+	Relation ResourceRelation `json:"-"`
+	// Access defines the type of access this resource requires.
+	Access runtime.Typed `json:"-"`
+	// Digest is the optional digest of the referenced resource.
+	Digest *Digest `json:"-"`
+	// Size of the resource blob.
+	Size int64 `json:"-"`
+	// CreationTime of the resource.
+	CreationTime CreationTime `json:"-"`
+}
+
+type CreationTime time.Time
+
+func (t CreationTime) DeepCopyInto(to *CreationTime) {
+	*to = t
+}
+
+// A Source is an artifact which describes the sources that were used to generate one or more of the resources.
+// Source elements do not have specific additional formal attributes.
+// See https://github.com/open-component-model/ocm-spec/blob/main/doc/01-model/02-elements-toplevel.md#sources
+// +k8s:deepcopy-gen=true
+type Source struct {
+	ElementMeta `json:",inline"`
+	Type        string        `json:"-"`
+	Access      runtime.Typed `json:"-"`
+}
+
+// Reference describes the reference to another component in the registry.
+// A component version may refer to other component versions by adding a reference to the component version.
+//
+// The Open Component Model makes no assumptions about how content described by the model is finally deployed or used.
+// This is left to external tools.
+//
+// Tool specific deployment information is formally represented by other artifacts with an appropriate type.
+//
+// In addition to the common artifact information, a resource may optionally describe a reference to the source by specifying its artifact identity.
+//
+// See https://github.com/open-component-model/ocm-spec/blob/main/doc/02-processing/01-references.md#referencing
+// +k8s:deepcopy-gen=true
+type Reference struct {
+	ElementMeta `json:",inline"`
```

Comment:
```suggestion
type Reference struct {
         // The name in ElementMeta describes the name of the reference itself within this component (not the name of the referenced component). But the version in ElementMeta specifies the version of the referenced component.
	ElementMeta `json:",inline"`
```
Since `Reference` is kind of weird with regards to `ElementMeta`, we might want to add an additional comment here.